### DISCUSSION TITLE
Alternatives to X-Accel-Redirect in a Kamal setup?

### DISCUSSION THREAD

**Discussion by jdelStrother** [basecamp/kamal#820] [Q&A]
I keep looking at switching our old capistrano/rails deployments to Kamal. One of the things I can't figure out a good replacement for is our use of nginx's X-Accel-Redirect to do an internal redirect to other services - eg our Rails app might respond with X-Accel-Redirect to a 127.0.0.1 URL pointing to an imgproxy service running on the same machine, which nginx then transparently fetches and the end-user just receives a regular 200 response. I don't see any support for this in Traefik or Thruster.

I guess I could leave our nginx layer in place, so traffic has to go through nginx -> traefik -> rails, but that seems like a lot of layers just to support an internal redirect. Any better suggestions?

**Comment by Splines** [comment]
see also: https://github.com/basecamp/kamal/discussions/232#discussioncomment-15282694
